### ISSUE TITLE
Terms: Enable Request Access - Incorrect info displayed in Terms tab of draft dataset

### ISSUE THREAD

**Issue by mheppler** [IQSS/dataverse#4829]
Tested on demo (4.9.1). Created a draft dataset, restricted files, added Terms of Access and checked "Enable access request"...

![screen shot 2018-07-10 at 3 46 49 pm](https://user-images.githubusercontent.com/687227/42533652-7edc56f0-8458-11e8-80fc-7c2b457ba85a.png)

Got the "Success! - The selected files have been updated." message. The green unlocked icon is now displayed next to the restricted file, but when you go to the dataset Terms tab, next to Request Access it reads, "Users may not request access to files."

![screen shot 2018-07-10 at 3 50 17 pm](https://user-images.githubusercontent.com/687227/42533833-fb951d80-8458-11e8-87ca-c857294bbaa2.png)

While this may be factually correct (users can not download a file from an unpublished draft dataset), it is not accurately displaying the proper setting selection to the depositor. The strangest thing though is that when you publish, the message changes to correctly read, "Users may request access to files."

This relates to a comment that @sbarbosadataverse made on issue #2902:

>as an admin or curator, when you set the request access feature, you can't see it next to the file name, until you log off and look at the dataset from the viewpoint of a user without a role in your dataverse. So it leaves you guessing if the feature was actually enabled

This is even more an egregious error than what Sonia points out, because it actually is misrepresented, as opposed to just plain missing.

**Comment by mheppler** [comment]
I was looking into this issue, and it seems that the solution is most likely in the backend logic. The HTML could be optimized though. Here is how the message gets to the page from `dataset-license-terms.xhtml`.

```
<div class="col-sm-9">
    <p>
        <h:outputText value="#{bundle['file.dataFilesTab.terms.list.termsOfAccess.requestAccess.request']}" rendered="#{DatasetPage.dataset.fileAccessRequest}"/>
        <h:outputText value="#{bundle['file.dataFilesTab.terms.list.termsOfAccess.requestAccess.notRequest']}" rendered="#{!DatasetPage.dataset.fileAccessRequest}"/>
    </p>
</div>
```
A better way to write that would be to put conditional render logic into one component, as opposed to two.

```
<h:outputText value="#{DatasetPage.dataset.fileAccessRequest ? bundle['file.dataFilesTab.terms.list.termsOfAccess.requestAccess.request'] : bundle['file.dataFilesTab.terms.list.termsOfAccess.requestAccess.notRequest']}"/>
```

In addition to a render logic fix that takes into account that the dataset is unpublished and the request access will be available to users, we could add a simple warning message to the UI, as mocked up below.

![screen shot 2018-07-17 at 10 38 03 am](https://user-images.githubusercontent.com/687227/42824365-b81d78e0-89ad-11e8-9f40-756c6523ef11.png)

**Comment by pdurbin** [comment]
I just created pull request #4860 and moved this issue to code review at https://waffle.io/IQSS/dataverse

**Comment by scolapasta** [comment]
One issue with this, which we have never handled well, is the fact that while you define these per version, the actual values that matter (and are used for the lock logic) are the published version (i.e. it gets pushed up to the dataset level when the version gets published). Which is probably what the logic is trying to accomplish. 

i.e. file request access is enabled in v1 then published; in v2 it is disabled and published. When we display v1, what should we should? 

My 2 cents is we should show that it had been enabled in v1, but is currently disabled. Showing just that is enabled is misleading.

**Comment by scolapasta** [comment]
So a little more feedback: the change in the PR changes from showing the version value instead of the dataset value. So this would change the way we display for older versions and is what is need of a better designed fix, as mentioned in my previous comment.

That said, I think we could do an incremental fix of showing the dataset value for now as we always do, unless it is a draft, and in that case, showing the version value. This is not perfect, but would leave the published versions as is, while still fixing the issues with drafts as was initially described above.

**Comment by pdurbin** [comment]
@scolapasta ok, I just opened pull request #4880 to implement the incremental fix you described above and closed the older pull request #4860.
